Edit username on desktop

janherich opened the pull request:
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Adds possibility to edit username on desktop - due to the desktop but, `text-input` has to have fixed size in order to be actually visible and editable, therefore, styling is little bit "meh" - when in edit mode, the username to be edit usually "jumps" little bit to the left or right depending on it's original length, but it does the job and we will fix the styling once underlying `text-input` bug is resolved. 

contribute to #5456 

### Testing notes (optional):
Test that username editing on desktop works

status: ready

janherich committed:
Edit username on desktop

Signed-off-by: yenda <eric@status.im>

vkjr commented:
@janherich, could you please add an issue in https://github.com/status-im/react-native-desktop and describe what should be fixed in `TextInput`?

<blockquote><img src="https://avatars3.githubusercontent.com/u/11767950?s=400&v=4" width="48" align="right"><div><img src="https://assets-cdn.github.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/status-im/react-native-desktop">status-im/react-native-desktop</a></strong></div><div>A Desktop port of React Native, driven by Qt, forked from Canonical - status-im/react-native-desktop</div></blockquote>

vkjr commented on src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:
```diff
@@ -15,14 +15,23 @@
+     [react/text-input {:font          :medium
+                        :font-size     18
+                        :default-value name
+                        :style         {:height             20
```
It would be nice to move styling in a separate file

churik commented:
Hey @janherich !
Thank you for your contribution.
Despite fact that `text-input` should be fixed in RN-size, I want to mention several issues here.

#### 1. Put focus to the editable area when the user is in editable mode
When you open Profile -> Edit you should manually set focus to username field. Now it is looking a bit confusing, especially when area is not distinguished from background.
I'm suggesting to set focus automatically to username area when you click on "Edit" (like it is made for mobile)
Video: http://take.ms/8A5ok

#### 2. When username is erased, "Type a message" is displayed
Expected: "Set username" or "Type a username" 
I believe @EugeOrtiz can provide design for this case.

For now it looks: 
![empty_us](https://user-images.githubusercontent.com/4557972/45751990-bbc29d80-bc1c-11e8-8c73-7f905175f457.png)

#### 3. Input area is not aligned to according to design.
See design  in zpl.io/brE1NdX
Now:
![px](https://user-images.githubusercontent.com/4557972/45752722-1e1c9d80-bc1f-11e8-8c8a-59a098b950e1.png)


#### 4. Font doesn't become italic like in design
Expected as it mentioned in zpl.io/brE1NdX
In fact, font is not changed when you change it.

#### 5. User remains in editable mode when he changes tab
Steps:
- Log in
- Open Profile
- Click on Edit
- Type new name in username field
- Click on Home
- Go Back to Profile

*Expected result:*
Edit button is available
*Actual result:* 
Done button is available; new name is replaced with previous one

#### 6. Username is cut in editable mode when it contains emoji
![emoji](https://user-images.githubusercontent.com/4557972/45753475-b287ff80-bc21-11e8-9947-a675b569306a.png)


Also I want to ask you to change "fixes" in description to "contributes", because this PR doesn't cover whole issue and provide link to RN-issue with `text-input` field.

Feel free to comment which issues can be created separately from your POV
As for me, 1, 2, 3 have higher priority.

janherich commented:
Thanks @churik !
I fixed the first issue by replacing the placeholder with empty string (the same as in mobile).
Regarding the second issue, I unfortunately could not fix it properly, as `text-input` on desktop doesn't react to `:auto-focus` property, so as a workaround, I changed the background to light grey when in edit mode.

churik commented:
@janherich could you please add an issue in https://github.com/status-im/react-native-desktop and describe what should be fixed in TextInput?

<blockquote><img src="https://avatars3.githubusercontent.com/u/11767950?s=400&v=4" width="48" align="right"><div><img src="https://assets-cdn.github.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/status-im/react-native-desktop">status-im/react-native-desktop</a></strong></div><div>A Desktop port of React Native, driven by Qt, forked from Canonical - status-im/react-native-desktop</div></blockquote>

janherich commented:
@churik it's already created - https://github.com/status-im/react-native-desktop/issues/338

churik commented:
1, 2 is fixed as mentioned in https://github.com/status-im/status-react/pull/5896#issuecomment-423071324.
All other issues and design for functionality will be fixed separately.

chadyj commented:
Great! Happy to see this merged and we can address other issues later.

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?
